cobc/pplex.l

```diff
@@ -1208,16 +1225,33 @@ ppopen (const char *name, struct cb_replace_list *replacing_list)
 	/* Check for BOM - *not* for input from stdin as rewind() clears the input
 	  buffer if used on stdin and output in console has normally no BOM at all */
 	if (ppin && strcmp (name, COB_DASH) != 0) {
-		unsigned char		bom[4];
-		if (fread (bom, 3, 1, ppin) == 1) {
-			if (bom[0] != 0xEF || bom[1] != 0xBB || bom[2] != 0xBF) {
-				rewind (ppin);
+		int fseek_to = 0 ;
+#define COBC_LOOKAHEAD 20
+		unsigned char buffer[COBC_LOOKAHEAD];
+		int nread = fread (buffer, 1, COBC_LOOKAHEAD, ppin);
+		int pos = 0;
+		if (nread >= 3 && buffer[0] == 0xEF && buffer[1] == 0xBB && buffer[2] == 0xBF) {
+			fseek_to = 3;
+			pos = 3;
+		}
+		if (source_format == CB_FORMAT_AUTO){
+			/* If indicator is wrong on first line, switch to free format */
+			/* skip empty lines */
+			while( nread-pos > 7 && ( buffer[pos] == '\r' || buffer[pos] == '\n' )){
+				pos++;
+			}
+			/* check tab or indicator */
+			if ( nread-pos > 7 && buffer[pos] != '\t' && !is_fixed_indicator (buffer[pos+6]) ){
+				fprintf(stderr, "Automatically switching to free format\n");
```

Comment:
that still needs to be a standard "note" with the current position (and  starting in lowercase)

Reply:
I added a warning that displays `free format detected` on the first line of the file.

Reply:
I'm sorry! I should have been more clear...

What about using 
```c
				struct cb_tree_common	loc;
    
				loc.source_file = name;
				loc.source_line = 1 + amount_of_0a_seen;
				loc.source_column = 7;
				cb_note_x (COB_WARNOPT_NONE, &loc, _("free format detected"));
```


and dropping the warning? Note: the test actually shows that the current line calculation is always 1, not the actual line.

Reply:
Ok, I dropped the warning, and added the note with correct line calculation. It's a bit a pity not to be able to disable the warning, though.

Reply:
> Ok, I dropped the warning, and added the note with correct line calculation.

Thank you.

>  It's a bit a pity not to be able to disable the warning, though.

There would be a way if you want to do that: add a new `-fformat=default` (which will be set in config/default.conf); this will be duplicating the current "auto" and then "auto" will not raise that note (only default will).